synapse/lib/auth.py

```diff
@@ -579,11 +579,28 @@ async def _addUser(self, iden, name):
         await self.feedBeholder('user:add', user.pack())
 
     async def addRole(self, name, iden=None):
+        '''
+        Add a Role to the Auth system.
+
+        Args:
+            name (str): The name of the role.
+            iden (str): A optional iden to use as the role iden.
+
+        Returns:
+            Role: A Role.
+        '''
         if self.roleidenbynamecache.get(name) is not None:
             raise s_exc.DupRoleName(mesg=f'Duplicate role name, {name=} already exists.', name=name)
 
         if iden is None:
             iden = s_common.guid()
+        else:
+            if not s_common.isguid(iden):
+                raise s_exc.BadArg(name='iden', arg=iden, mesg='Argument it not a valid iden.')
```

Comment:
Since Optic displays the `mesg` key to users, we like to tuck the user provided variables into the message when possible. Can add `{iden}` to the message and make it a fstring. Similar to the DupIden mesg as well.

Reply:
We still want to retain the additional keys though - with structured logging those values can become values that are indexed / searchable by logging solutions.